cmd/nomos/status/client.go

```diff
@@ -255,10 +255,16 @@ func (c *ClusterClient) multiRepoClusterStatus(ctx context.Context, cs *ClusterS
 		}
 	}
 
-	if len(rootSyncs) != 0 {
+	if len(rootSyncs) != len(rootRGs) {
+		errs = append(errs, fmt.Sprintf("expected the number of RootSyncs and ResourceGroups to be equal, but found %d RootSyncs and %d ResourceGroups", len(rootSyncs), len(rootRGs)))
+	} else if len(rootSyncs) != 0 {
 		var repos []*RepoState
 		for i, rs := range rootSyncs {
 			rg := rootRGs[i]
+			if rg == nil {
+				// We always expect a ResourceGroup, even if we have no managed resources.
```

Comment:
I imagine it's possible that there's no RG if there was an error before it was created. Will this suppress or expose that kind of error?

Reply:
If there is no RG because of an error before it was created, we should still report that the RG is missing in `nomos status` right? We should always have a ResourceGroup per R*Sync I belive.

Reply:
Previously when nomos CLI used a GET method, we'd always surface errors when we can't find the ResourceGroup. A regression occurred when we updated the status command to LIST all ResourceGroups, thus we didn't surface that the ResourceGroup is missing after that change was merged.